cmds/cpu/cpu9p.go

```diff
@@ -144,8 +145,21 @@ func (l *cpu9p) ReadAt(p []byte, offset int64) (int, error) {
 }
 
 // Write implements p9.File.WriteAt.
+// There is a very rare case where O_APPEND files are written more than
+// once, and we get an error. That error is generated by the Go runtime,
+// after checking the open flag in the os.File struct.
+// I.e. the error is not generated by a system call,
+// so it is very cheap to try the WriteAt, check the
+// error, and call Write if it is the rare case of a second write
+// to an append-only file..
 func (l *cpu9p) WriteAt(p []byte, offset int64) (int, error) {
-	return l.file.WriteAt(p, int64(offset))
+	n, err := l.file.WriteAt(p, int64(offset))
+	if err != nil {
+		if strings.Contains(err.Error(), "os: invalid use of WriteAt on file opened with O_APPEND") {
```

Comment:
it's a very Plan9 thing to treat error strings as ABI.  =)

does Go treat error strings as a stable ABI?  if not, if we ever have easy tests for cpu, we can make sure the O_APPEND case is tested so we can catch the breakage.

Reply:
yep, it's a problem. Few people know or care however :-(